CauseContract.sol

```diff
@@ -25,23 +41,43 @@ contract CauseContract {
         uint256 gasUsed;
         uint256 transactionFee;
     }
-
+    
     // incoming donations
     Transaction[] incoming;
 
     // outgoing funding
     Transaction[] outgoing;
 
+    // donor proportion tracking
+    mapping(address => uint256) public donorTotals;
+
+
+    // endCause flag -> 1 = False, 2 = True
+    uint256 public endCause = 1;
+
     // contract information struct
     struct ContractInfo {
         string id;
         address admin;
         Transaction[] incoming;
         Transaction[] outgoing;
         address contractAddress;
+        uint256 causeTotal;
+        uint256 endCause;
```

Comment:
Ah, one last thing could you change endCause to causeState just so its clearer when integrating the front-end, ty